pkg/rancher-components/src/components/Form/Checkbox/Checkbox.vue

```diff
@@ -169,6 +169,12 @@ export default Vue.extend({
           addObject(this.value, this.valueWhenTrue);
         }
         this.$emit('input', this.value);
+      } else if (this.isString(this.valueWhenTrue)) {
+        if (this.isChecked) {
+          this.$emit('input', undefined);
```

Comment:
I'm not sure how I feel about `undefined` to represent the absence of a value in this case.. maybe `null` or an empty string would be more appropriate. Neither feel great to me, what do you think?

We also don't have the concept of a `valueWhenFalse` for this component. Looking into these change makes me feel strongly about the need to revisit this component as a whole further down the line.

Reply:
I initially went with `null` for the absence of a value, but decided on `undefined` as it was the default value for the resources of a container. But I think `null` would be the best solution for now.
Having a `valueWhenFalse` would be beneficial for sure, especially in this case as the false values for `httpEndpoint` and `httpTokens` are intended to be `disabled` and `optional` respectively. It's not entirely necessary for these properties but definitely worth a revisit.

Reply:
I can see how the lack of a `valueWhenFalse` might be problematic when certain data might expect a different value than 'null`.. We run the risk of overwriting expected values for fields that might expect a custom value for a `false` by only providing `null` when `valueWhenTrue` is in use.

I updated #6787 with some notes about this for us to consider when we look at a refactor.